src/github.js

```diff
@@ -0,0 +1,38 @@
+const path = require("path");
+const fetch = require('make-fetch-happen');
+
+module.exports = class GithubAPI {
+  cacheDir;
+  auth;
+
+  constructor(config = {}) {
+    config.cacheDir = '.changelog';
+    this.cacheDir = config.cacheDir && path.join(process.cwd(), config.cacheDir, "github");
+    this.auth = this.getAuthToken();
+    if (!this.auth) {
+      throw new Error("Must provide GITHUB_AUTH as a ENV variable");
+    }
+  }
+
+  async getIssueData(repo, issue) {
+    return this._fetch(`https://api.github.com/repos/${repo}/issues/${issue}`);
+  }
+
+  async _fetch(url) {
+    const res = await fetch(url, {
+      cacheManager: this.cacheDir,
+      headers: {
+        Authorization: `token ${this.auth}`,
+      },
+    });
+    const parsedResponse = await res.json();
+    if (res.ok) {
+      return parsedResponse;
+    }
+    throw new ConfigurationError(`Fetch error: ${res.statusText}.\n${JSON.stringify(parsedResponse)}`);
+  }
+
+  getAuthToken() {
+    return process.env.GITHUB_AUTH || "";
+  }
```

Comment:
why is a method needed here? it's only called once in the constructor

Reply:
it's not 😂 I've removed it and inlined the code in the constructor 👍